enhancements/etcd/protecting-etcd-quorum-during-control-plane-scaling.md

```diff
@@ -0,0 +1,317 @@
+    * This also ensures that we keep the full Etcd data on disk on a minimum of the desired number of Control Plane Machines at all times
+
+#### Protecting Etcd during scaling operations
+
+To ensure that the safety requirements described above are maintained during scaling operations, the Etcd Operator will manage the Etcd cluster in clusters with a [functional Machine API](#When-is-Machine-API-Functional), by leveraging [Machine Deletion Hooks](#What-are-Machine-Deletion-Hooks) to coordinate with the Machine API when it is, and isn't safe, to remove Machines with voting members of the Etcd cluster running on them.
+
+##### Overview of protecion mechanism
+
+To ensure the safety of the Etcd cluster quorum, the Etcd Operator will leverage a pre-drain [Machine Deletion Hooks](#What-are-Machine-Deletion-Hook) to prevent the removal of any Control Plane Machine hosting a voting member of the Etcd quorum.
+
+```yaml
+pre-drain.delete.hook.machine.cluster-api.x-k8s.io/etcd-quorum-member: openshift-etcd-operator
+```
+
+The Etcd Operator will apply the hook to a Machine resource once it identifies that the Machine hosts an Etcd quorum member.
+
+It will only remove the hook once it has identifed that the Machine resource is being deleted and a replacement member has been created. The removal of this hook will allow the Machine API to drain and terminate the Machine as it would normally do.
+
+The Etcd Operator will ensure, based on the desired Control Plane replica count in <@Sam where do we get this from today?>, that the Etcd cluster has either the exact desired count of voting members, or during scaling/replacement operations, at most 1 extra voting member.
+
+The Etcd Operator will also leverage the Etcd quorum guard to prevent voluntary disruptions of Etcd members during the process.
+By ensuring that the quorum guard PDB always has `minAvailable: (Num Current Control Plane Machines) - 1`, this prevents draining of a healthy Etcd member until a new member becomes healthy. This operation will prevent other components in the cluster (eg. MCO) from disrupting the quorum of Etcd during this operation.
+
+##### Adoption of an existing Control Plane into the new mechanism
+
+Note: This flow is the expectation for when a cluster is upgraded from 4.N-1 to 4.N where 4.N is the version where this mechanism is introduced.
+
+1. Etcd Operator fetches all Control Plane Machines
+1. Etcd Operator identifies if the Control Plane Machines are in the `Running` phase
+    a. If no Control Plane Machines are `Running`, assume that Machine API is non-functional and stop here
+1. Etcd Operator identifies the voting members of the Etcd cluster and maps these to the Control Plane Machines that host them
+1. Etcd Operator adds a deletion hook to each of the Machines hosting a voting member of the Etcd cluster
+
+##### Operational flow during a resize operation
```

Comment:
Probably a nit:
```suggestion
##### Operational flow during a node replace operation
```
IIUC this flow describes a node replacement operation (via scale up then down) rather than a resize operation as the final cluster size is unchanged.
Although I guess this could be a resize if you only add or remove a machine.

Reply:
The context behind this was a resize of the underlying machine, ie, vertical scaling of the host. But I can see how that can be confusing from an etcd perspective, so I'll adjust the wording